pandas/core/generic.py

```diff
@@ -8327,24 +8327,95 @@ def _doc_parms(cls):
 """
 
 _cnum_doc = """
+Return %(desc)s over a DataFrame or Series axis.
+
+Returns a DataFrame or Series of the same size containing the %(desc)s.
 
 Parameters
 ----------
 axis : %(axis_descr)s
 skipna : boolean, default True
     Exclude NA/null values. If an entire row/column is NA, the result
-    will be NA
+    will be NA.
+*args : any, default None
+**kwargs : any, default None
+    Additional keywords have no effect but might be accepted for
+    compatibility with NumPy.
 
 Returns
 -------
 %(outname)s : %(name1)s\n
-
-
+%(examples)s
 See also
 --------
 pandas.core.window.Expanding.%(accum_func_name)s : Similar functionality
     but ignores ``NaN`` values.
+pandas.Series.%(outname)s : Return %(desc)s over Series axis.
+pandas.DataFrame.cummax : Return cumulative maximum over DataFrame axis.
+pandas.DataFrame.cummin : Return cumulative minimum over DataFrame axis.
+pandas.DataFrame.cumsum : Return cumulative sum over DataFrame axis.
+pandas.DataFrame.cumprod : Return cumulative product over DataFrame axis.
+"""
 
+_cummax_examples = """\
+Examples
+--------
+**DataFrame**
+
+Create a DataFrame:
+
+>>> df = pd.DataFrame([[9, 7, 9, 7],
+...                    [7, 5, 2, 7],
+...                    [3, 5, 2, 2],
+...                    [8, 0, 9, 0]],
+...                    columns=list('ABCD'))
```

Comment:
I'd use a much smaller dataset, probably 2 columns and 3 or 4 rows. Smaller numbers would make it easier for users to see what's being added, specially if we reuse same dataframe for `cumsum`...

Also, you can add a `NaN`, so you can show an example with `skipna`.